model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/common/HotRodEntityDescriptor.java

```diff
@@ -17,35 +17,34 @@
 
 package org.keycloak.models.map.storage.hotRod.common;
 
-import java.util.List;
-import java.util.stream.Stream;
+import org.keycloak.models.map.storage.ModelEntityUtil;
 
-public class HotRodEntityDescriptor<EntityType> {
+import java.util.function.Function;
+
+public class HotRodEntityDescriptor<E, D extends HotRodEntityDelegate<E>> {
     private final Class<?> modelTypeClass;
-    private final Class<EntityType> entityTypeClass;
-    private final List<Class<?>> hotRodClasses;
-    private final String cacheName;
+    private final Class<E> entityTypeClass;
+    private final Function<E, D> hotRodDelegateProvider;
 
-    public HotRodEntityDescriptor(Class<?> modelTypeClass, Class<EntityType> entityTypeClass, List<Class<?>> hotRodClasses, String cacheName) {
+    public HotRodEntityDescriptor(Class<?> modelTypeClass, Class<E> entityTypeClass, Function<E, D> hotRodDelegateProvider) {
```

Comment:
There should be either `modelTypeClass` or `entityTypeClass` (perhaps the latter). The other one can be obtained from [`ModelEntityUtil`](https://github.com/keycloak/keycloak/blob/95614e8b4022306ffac328b85130403cc157fb18/model/map/src/main/java/org/keycloak/models/map/storage/ModelEntityUtil.java#L116)

Reply:
I am afraid that is not possible. `ModelEntityUtil` doesn't contain reference to HotRod entities, like `HotRodClientEntity`.

See: 
https://github.com/keycloak/keycloak/blob/7baaddd99dff29b139789009c9951c2f06a702c6/model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/HotRodMapStorageProviderFactory.java#L55-L58